src/app/authentication/components/form-login/form-login.component.ts

```diff
@@ -1,13 +1,49 @@
 import { Component, OnInit } from '@angular/core';
+import { AuthenticationService } from '../../service/authentication.service';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { Router } from '@angular/router';
+import { AuthService } from 'src/app/shared/services/auth.service';
 
 @Component({
   selector: 'app-form-login',
   templateUrl: './form-login.component.html',
   styleUrls: ['./form-login.component.scss']
 })
 export class FormLoginComponent implements OnInit {
+  loginForm: any;
+  errorMessage = '';
 
-  constructor() {}
+  constructor(
+    private authenticationService: AuthenticationService,
+    private authService: AuthService,
+    private router: Router
+  ) {
+    if (authService.isLoggedIn()) {
+      router.navigate(['/']);
+    }
+  }
 
-  ngOnInit(): void {}
+  ngOnInit(): void {
+    this.loginForm = new FormGroup({
+      email: new FormControl('', [Validators.required, Validators.email]),
+      password: new FormControl('', [Validators.required])
+    });
+  }
+
+  onSubmit(): void {
+    this.errorMessage = '';
+
+    this.authenticationService.login(this.loginForm.value.email, this.loginForm.value.password).subscribe(
+      response => {
+        this.authService.setAccessToken(response.accessToken, response.tokenType);
+
+        this.router.navigate(['/']);
+      },
+      error => {
+        if (error.status === 400) {
```

Comment:
@carryall Could we map http status codes to errors massage? 🤔 

```js
const ERROR_MESSAGES = {
  200: 'Invalid email or password'
 ....
 ....
}

```

 `this.errorMessage = ERROR_MESSAGES[error.status]`

Reply:
@junan Updated in [df40770](https://github.com/carryall/survey-challenge/pull/29/commits/df4077032fad694365c517cb0fb49cb608dae6fa)